### PR TITLE
Switch from `env_logger` to `tracing`

### PR DESCRIPTION
**Pull request by wkozyra95**
- Use tracing library
- Default to json logger when not building from source (based on CARGO_MANIFEST_DIR env)
- Add env `FFMPEG_LOG_LEVEL` that controll ffmpeg logs (default to warn)

### CODE REVIEW

**Review by noituri — COMMENTED** [review]
It doesn't compile on MacOS
image

File: src/logger.rs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            FfmpegLogLevel::Debug if log_level <= 48 => {
+                ffmpeg_next::sys::av_log_default_callback(arg1, log_level, fmt, va_list_tag)
+            }
+            _ => (),
+        }
+    }
+}
+
+pub fn init_logger() {
+    if env::var("CARGO_MANIFEST_DIR").is_ok() {
```
Comment (jerzywilczek) [inline src/logger.rs:50]: Shouldn't we also have a manual switch for this?

**Review by jerzywilczek — APPROVED** [review]
:rocket:

File: src/logger.rs (hunk 1 above)
Comment (wkozyra95) [reply src/logger.rs:50]: I'll be reworking env based configuration in follow-up PR, so I can address it then.

File: src/logger.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            }
+            FfmpegLogLevel::Info if log_level <= 32 => {
+                ffmpeg_next::sys::av_log_default_callback(arg1, log_level, fmt, va_list_tag)
+            }
+            FfmpegLogLevel::Debug if log_level <= 48 => {
+                ffmpeg_next::sys::av_log_default_callback(arg1, log_level, fmt, va_list_tag)
+            }
+            _ => (),
+        }
+    }
```
Comment (WojciechBarczynski) [inline src/logger.rs:49]: I had a hard time understanding, that `log_level` refers to value received from FFmpeg, while result of `ffmpeg_log_level` is value set by user. Changing `ffmpeg_log_level` to `ffmpeg_logging_level` would solve that.

File: src/logger.rs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    static LOG_LEVEL: OnceLock<FfmpegLogLevel> = OnceLock::new();
+
+    *LOG_LEVEL.get_or_init(|| match env::var("FFMPEG_LOG_LEVEL") {
+        Ok(debug) if debug == "debug" => FfmpegLogLevel::Debug,
+        Ok(info) if info == "info" => FfmpegLogLevel::Info,
+        Ok(warn) if warn == "warn" => FfmpegLogLevel::Warn,
+        Ok(error) if error == "error" => FfmpegLogLevel::Error,
+        Ok(_) => FfmpegLogLevel::Warn,
+        Err(_) => FfmpegLogLevel::Warn,
+    })
```
Comment (WojciechBarczynski) [inline src/logger.rs:21]: Do we want to use env vars for such configurations?
We have more and more things configurable by env vars in similar way. Isn't it better to use some config file instead?

Comment (wkozyra95) [reply src/logger.rs:21]: I addressed that in https://github.com/membraneframework/video_compositor/pull/342 description

Comment (WojciechBarczynski) [reply src/logger.rs:21]: That explanation convinced me 👍

**Pull request merged**